pkg/network/tracer/tracer_linux_test.go

```diff
@@ -69,13 +69,14 @@ func doDNSQuery(t *testing.T, domain string, serverIP string) (*net.UDPAddr, *ne
 	queryMsg.SetQuestion(dns.Fqdn(domain), dns.TypeA)
 	queryMsg.RecursionDesired = true
 	dnsClient := new(dns.Client)
-	dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
-	require.NoError(t, err)
-	defer dnsConn.Close()
-	dnsClientAddr := dnsConn.LocalAddr().(*net.UDPAddr)
-	_, _, err = dnsClient.ExchangeWithConn(queryMsg, dnsConn)
-	require.NoError(t, err)
-
+	var dnsClientAddr *net.UDPAddr
+	require.Eventually(t, func() bool {
+		dnsConn, err := dnsClient.Dial(dnsServerAddr.String())
+		dnsClientAddr = dnsConn.LocalAddr().(*net.UDPAddr)
+		_, _, connErr := dnsClient.ExchangeWithConn(queryMsg, dnsConn)
```

Comment:
Would this panic or otherwise misbehave if `dnsClient.Dial` returns an err above?

Reply:
good question... I haven't see that in practice but can write this a bit safer

Reply:
Lets do that please. There is a version of `Eventually` that passes in a context that can be used to indicate failure. https://pkg.go.dev/github.com/stretchr/testify/assert#EventuallyWithT

Reply:
changed this block to use `EventuallyWithT` and assert that the two calls here pass individually. if the first assertion fails, we bail out and the eventually will start over on the next "tick"

Reply:
Similar to my comment in https://github.com/DataDog/datadog-agent/pull/20347/files#diff-e2871eefa3722b7a4b0bb1234dbbb5bb12305f81428382219b1b5d4ba5a67095R1062 , I'm concerned that this will trigger timeouts.

I had a half-done branch which created a local DNS server that should be more reliable than 8.8.8.8, but would love a cleaner solution than that. 

https://github.com/DataDog/datadog-agent/pull/20141/files